server/WebAPI/Controllers/ScheduledPlaceToEatController.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using WebAPI.DTO.ScheduledPlaceToEat;
+using WebAPI.Extension;
+using WebAPI.Interfaces;
+using WebAPI.Services.Assets;
+
+namespace WebAPI.Controllers
+{
+  [Authorize]
+  [Route("api/[controller]")]
+  [ApiController]
+  public class ScheduledPlaceToEatController : ControllerBase
+  {
+    private readonly IScheduledPlaceToEatService _scheduledPlaceToEatService;
+    private readonly IAttachmentFileEatingService _attachmentFileEatingService;
+
+    public ScheduledPlaceToEatController(
+      IScheduledPlaceToEatService scheduledPlaceToEatService,
+      IAttachmentFileEatingService attachmentFileEatingService)
+    {
+      _scheduledPlaceToEatService = scheduledPlaceToEatService;
+      _attachmentFileEatingService = attachmentFileEatingService;
+    }
+
+    [HttpGet]
+    [Route("{id}")]
+    [ProducesResponseType(typeof(InputScheduledPlaceToEatForCreateDTO), StatusCodes.Status200OK)]
+    public IActionResult GetById([FromRoute] int id)
+    {
+      return Ok(_scheduledPlaceToEatService.GetEatingByUserId(HttpContext.GetUserIdFromClaim()).Where(r => r.Id.Equals(id)).FirstOrDefault());
+    }
+
+    // GET: api/ScheduledPlaceToEat
+    [HttpGet]
+    [ProducesResponseType(typeof(IEnumerable<InputScheduledPlaceToEatForCreateDTO>), StatusCodes.Status200OK)]
+    public IActionResult Get()
+    {
+      return Ok(_scheduledPlaceToEatService.GetEatingByUserId(HttpContext.GetUserIdFromClaim()));
+    }
+
+    [HttpPost("UploadEatingMultiFile/{id}")]
+    [Consumes("multipart/form-data")]
+    public IActionResult UploadEatingMultiFile([FromForm] IFormFileCollection files, int id)
+    {
+      if(files.Count <= Consts.MaxEatingFileCount && files.Count > 0)
+      {
+        foreach(var file in files)
+        {
+          if(!Consts.AllowedImageContentTypes.Contains(file.ContentType, StringComparer.OrdinalIgnoreCase) && !Consts.AllowedTextContentTypes.Contains(file.ContentType, StringComparer.OrdinalIgnoreCase))
```

Comment:
might you need model validators?

Reply:
I could not make the server accept at least some file, so the model validator I don't think I can do it

Reply:
might, you try?
https://stackoverflow.com/questions/46694745/asp-net-core-custom-model-validation

Reply:
ok